unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/BaseLayer.cs

```diff
@@ -100,6 +101,127 @@ public TextureLayer(string name = "noname")
 
             particleRotation = new List<TransitioningVector3>();
         }
+
+        // Copy details of present layer on the passed layer
+        public void DeepCopy(TextureLayer layer)
+        {
+            layer.slotID = this.slotID;
```

Comment:
Can we use `MemberwiseClone` for primitives and only do the deepcopy for reference types? This approach will lower the high maintenance cost of this method.

Reply:
done